target/linux/ramips/mt7621/base-files/lib/upgrade/iodata.sh

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright (C) 2019 OpenWrt.org
+#
+
+. /lib/functions.sh
+
+iodata_mstc_prepare_fail() {
```

Comment:
IMO device-specific upgrade procedures do not need their own reboot sequence, it should be covered in [do_stage2](https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/upgrade/do_stage2). But [nand.sh](https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/upgrade/nand.sh) and [zyxel.sh](https://github.com/openwrt/openwrt/blob/master/target/linux/ipq806x/base-files/lib/upgrade/zyxel.sh) also have their own reboot commands, and currently do_stage2 does not utilize the return value of `platform_do_upgrade()`.
So I'm okay with `iodata_mstc_prepare_fail()` at this moment, but I wish we can remove it in the future. I'll try investigating it when I have time.